src/main/java/com/browserstack/automate/ci/common/BrowserStackBuildWrapperOperations.java

```diff
@@ -129,6 +130,14 @@ public void buildEnvVars(Map<String, String> env) {
             }
         }
 
+        if ( env.get(BrowserStackEnvVars.GRR_JENKINS_KEY) != null && !env.get(BrowserStackEnvVars.GRR_JENKINS_KEY).isEmpty() && Constants.GRR_AUTO_REGION_VS_APIURL.containsKey(env.get(BrowserStackEnvVars.GRR_JENKINS_KEY).toLowerCase()) ) {
```

Comment:
[Not a blocker] A cleaner way would be to move `env.get(BrowserStackEnvVars.GRR_JENKINS_KEY)` to a variable and use it everywhere to make it more readable.